deepmd/dpmodel/descriptor/make_base_descriptor.py

```diff
@@ -84,8 +86,13 @@ def mixed_types(self) -> bool:
             """
             pass
 
+        @abstractmethod
+        def share_params(self, base_class, shared_level, resume=False):
+            """Share the parameters of self to the base_class with shared_level."""
+            pass
+
         def compute_input_stats(
-            self, merged: List[dict], path: Optional[DPPath] = None
+            self, merged: Union[Callable, List[dict]], path: Optional[DPPath] = None
```

Comment:
What is the purpose of this change?

Reply:
Anyway, the type `Callable` should be more detailed by giving its input and output types.

Reply:
> What is the purpose of this change?

To pack data for statistics is a very slow and memory-block process, especially when multitask training. Here I used one lazy function to do this, because only inside descriptor&fitting, one can finally decide whether to do data stat or not. If do stat, one can call this lazy function and pack the data for statistics for just once in whole process.

Reply:
> Anyway, the type `Callable` should be more detailed by giving its input and output types.

Updated.

Reply:
fyi @wanghan-iapcm